src/zarr/testing/strategies.py

```diff
@@ -232,14 +238,14 @@ def np_array_and_chunks(
     return (array, draw(chunk_shapes(shape=array.shape)))
 
 
-@st.composite  # type: ignore[misc]
+@st.composite
 def arrays(
     draw: st.DrawFn,
     *,
     shapes: st.SearchStrategy[tuple[int, ...]] = array_shapes,
     compressors: st.SearchStrategy = compressors,
     stores: st.SearchStrategy[StoreLike] = stores,
-    paths: st.SearchStrategy[str | None] = paths(),  # noqa: B008
+    paths: st.SearchStrategy[str] = paths(),  # noqa: B008
```

Comment:
None should be allowed here, no?

Reply:
Shouldn't paths just be strings? At least `Group.create_array()` (where this variable is passed in the tests) doesn't accept `path=None`.